src/Accord.Bot/Infrastructure/ReminderService.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Remora.Discord.API.Abstractions.Rest;
+using Remora.Discord.Core;
+
+namespace Accord.Services.Reminder
+{
+    public class ReminderService : BackgroundService
+    {
+        private readonly IServiceScopeFactory _serviceScopeFactory;
+
+        public ReminderService(IServiceScopeFactory serviceScopeFactory)
+        {
+            _serviceScopeFactory = serviceScopeFactory;
+        }
+
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            using var scope = _serviceScopeFactory.CreateScope();
+            var services = scope.ServiceProvider;
+
+            var mediator = services.GetRequiredService<IMediator>();
+
+            var channelApi = services.GetRequiredService<IDiscordRestChannelAPI>();
+
+            await ProcessReminders(mediator, channelApi, true, stoppingToken);
+
+            while (!stoppingToken.IsCancellationRequested)
+            {
+                await ProcessReminders(mediator, channelApi, false, stoppingToken);
+            }
+        }
+
+        private async Task ProcessReminders(IMediator mediator, IDiscordRestChannelAPI channelApi, bool discard, CancellationToken stoppingToken)
+        {
+            var reminders = await mediator.Send(new GetAllRemindersRequest(), stoppingToken);
+
+            var processableReminders = reminders.Value!.Where(x => x.RemindAt <= DateTime.Now);
+
+            foreach (var reminder in processableReminders)
+            {
+                if ((discard && (DateTime.Now - reminder.RemindAt).Minutes < 1) || !discard)
+                {
+                    await channelApi.CreateMessageAsync(new Snowflake(reminder.DiscordChannelId),
+                        $"[Reminder] <@{reminder.UserId}> -- {reminder.Message}", ct: stoppingToken);
```

Comment:
People could potentially abuse bot permissions here, likely be best if we captured this into an embed and pinged the owner of the reminder, i.e.:

```
@JohnDoe
{Embed}
```

Reply:
sure done